src/components/ReadmePage.tsx

```diff
@@ -86,7 +86,9 @@ class ReadmePage extends React.Component<
         <div className="marketplace-header">
           <h1>{this.props.title}</h1>
         </div>
-        <div id="marketplace-readme" className="marketplace-readme__container" dangerouslySetInnerHTML={{ __html: this.state.html }}></div>
+        {this.state.html === "<p>Loading...</p>"
+          ? <footer className="marketplace-footer"><LoadingIcon /></footer>
+          : <div id="marketplace-readme" className="marketplace-readme__container" dangerouslySetInnerHTML={{ __html: this.state.html }}></div>}
```

Comment:
I wonder if it's possible to just make the `state.html` the loading markup. Or maybe not because it's JSX and the html needs to be a string?

Reply:
I tried it before, but it just returns `[object Object]` the exact string, sometimes it won't even update.

Reply:
Ah yeah, that makes sense. It's trying to convert the loading icon component to a string, so the best it can do is output that it's an object. Makes sense. I briefly looked into if we could make the page itself JSX, but it doesn't look like that's super easy or worth it.